### ISSUE TITLE
Cattle cluster agent fails to resolve rancher hostname

### ISSUE THREAD

**Issue by korenlev** [rancher/rancher#33272]
Duplicate of:
https://github.com/rancher/rancher/issues/29780

In our production case we use aws route53 dns record with wildcard to route *c1.domain.com to clstr1 and *c2.domain.com to clstr2 etc. Nodes are aws instances.
This way we do not need 1M records for each clstr services, so for example rancher resolves using rancher.c1.domain.com (hits the * wilcard).
We have mutiple clusters works fine till 2.5., and now bc of this bug we can not install rancher.
From any host or pod a dns ping to rancher fqdn returns pong,    only cattle cluster agent can't resolve it (aka: the bug)

**Comment by korenlev** [comment]
Cattle cluster agent may be hitting this old easily fixed bug:
https://github.com/coredns/coredns/issues/776

**Comment by superseb** [comment]
Please supply exact v2.5 version used, and what v2.4 version you used + k8s version that worked. If there is a regression, we can look into it but we need the exact version so we know what changed. Having a wildcard DNS is basically the same as what is described in https://github.com/rancher/rancher/issues/29780#issuecomment-745166483

**Comment by korenlev** [comment]
issue seen in clusters with : 2.5.7, 2.5.8

issue not seen in clusters with: 2.3.5, 2.4.8

**Comment by superseb** [comment]
Please share the requested information, what k8s version were you running in those versions and if possible, the logging from the CoreDNS pods in the working version? You didn't upgrade the OS during the time between those versions?

**Comment by korenlev** [comment]
K8s version 1.19 and moving to 1.20 (with the unsuccessful deploy of 2.5.x)
btw - 2.5.6 tested - same issue.
no upgrade to OS, still using ubuntu 18.0.4 on the nodes.
as we are reverting back to 2.4.x , i will try on testing environment soon to provide the cattle cluster agent logs

**Comment by korenlev** [comment]
on the working versions, coreDNS just shoots several timeout ERRORS:

[ERROR] plugin/errors: 2 rmq.this.domain.com. AAAA: read udp 10.x.y.13:45935->10.0.3.85:53: i/o timeout

multiple times .....
(real values masked)

**Comment by superseb** [comment]
Yeah, k8s version used with 2.4.x would help (exact version) so I can look up the add-on versions and start there. Also please post full logs and not just a single line.

**Comment by TheRealAlexV** [comment]
Can we re-open this? I'm also having this issue. 

Rancher version 2.6.0
kubernetes version v1.20.11-rancher1-2
Ubuntu 20.04 nodes
3 workers, 3 masters

I checked the logs of the cattle-cluster-agent and found that it's having issues resolving the hostname for the cattle server:
```
INFO: Environment: CATTLE_ADDRESS=10.42.1.2 CATTLE_CA_CHECKSUM=***
CATTLE_CLUSTER=true CATTLE_CLUSTER_AGENT_PORT=tcp://10.43.215.205:80 CATTLE_CLUSTER_AGENT_PORT_443_TCP=tcp://10.43.215.205:443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_ADDR=10.43.215.205 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PORT=443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_PORT_80_TCP=tcp://10.43.215.205:80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_ADDR=10.43.215.205 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PORT=80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_SERVICE_HOST=10.43.215.205 CATTLE_CLUSTER_AGENT_SERVICE_PORT=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTP=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTPS_INTERNAL=443 CATTLE_FEATURES=embedded-cluster-api=false,fleet=false,monitoringv1=false,multi-cluster-management=false,multi-cluster-management-agent=true,provisioningv2=false,rke2=false CATTLE_INTERNAL_ADDRESS= CATTLE_IS_RKE=true CATTLE_K8S_MANAGED=true CATTLE_NODE_NAME=cattle-cluster-agent-754bb774c7-n6tt2 CATTLE_SERVER=https://rancher.mydomain.com
INFO: Using resolv.conf: nameserver 10.43.0.10 search cattle-system.svc.cluster.local svc.cluster.local cluster.local mydomain.com options ndots:5
ERROR: https://rancher.mydomain.com/ping is not accessible (Could not resolve host: rancher.mydomain.com)
```

Thing is, all 3 of my masters and all 3 of my workers can resolve the domain just fine. CoreDNS pods just show some io timeout errors:
```
linux/amd64, go1.15.3, 054c9ae
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
I0924 16:47:56.859902       1 trace.go:205] Trace[1427131847]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156 (24-Sep-2021 16:47:26.859) (total time: 30000ms):
Trace[1427131847]: [30.000324176s] [30.000324176s] END
I0924 16:47:56.859923       1 trace.go:205] Trace[911902081]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156 (24-Sep-2021 16:47:26.859) (total time: 30000ms):
Trace[911902081]: [30.000158123s] [30.000158123s] END
E0924 16:47:56.859925       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Namespace: failed to list *v1.Namespace: Get "https://10.43.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
E0924 16:47:56.859933       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Service: failed to list *v1.Service: Get "https://10.43.0.1:443/api/v1/services?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
I0924 16:47:56.859937       1 trace.go:205] Trace[939984059]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156 (24-Sep-2021 16:47:26.859) (total time: 30000ms):
Trace[939984059]: [30.000156648s] [30.000156648s] END
E0924 16:47:56.859941       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Endpoints: failed to list *v1.Endpoints: Get "https://10.43.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
```

Does Coredns forward it's dns requests to the DNS servers that the host has configured? It would appear so judging by the logs output, but it's not behaving that way.

I don't have any wildcards involved in my DNS setup yet, however I am doing split horizon DNS, where rancher.mydomain.com resolves to the public or private facing IP depending on what side of the network you're on. It resolve correctly when tested with nslookup from the host.

**Comment by xyonium** [comment]
@TheRealAlexV did you solve this problem, I'm facing similar problem.
